lib/core/src/main/java/io/atlasmap/v2/StoreOutput.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atlasmap.v2;
+
+import io.atlasmap.api.AtlasFieldAction;
+import io.atlasmap.spi.AtlasFieldActionInfo;
+import io.atlasmap.spi.AtlasFieldActionParameter;
+
+/**
+ * Stores the output of the previous action in a variable.
+ */
+public class StoreOutput extends Action implements AtlasFieldAction {
+
+    private static final long serialVersionUID = 1L;
+
+    @AtlasFieldActionParameter()
+    private String variable;
+
+    @AtlasFieldActionInfo(name = "StoreOutput", sourceType = FieldType.ANY, targetType = FieldType.ANY, sourceCollectionType = CollectionType.ALL, targetCollectionType = CollectionType.ALL)
+    public static Object execute(StoreOutput action, Object input) {
+        return input;
```

Comment:
This looks incomplete, can you remove this and file a separate issue for adding `StoreOutput`?

Reply:
I can, but I was using it here because this leads towards the variable support I'm ultimately providing. If you like, I can remove it though and just provide unit tests alone with a test transformation for this.

Reply:
Again, keep your personal preparation in your private branch